src/typegen.ts

```diff
@@ -6,12 +6,14 @@ import { AppOptions } from './types'
 import { Program } from 'typescript'
 import { version } from '../package.json'
 
-// The undocumented defaultMaximumTruncationLength setting determines at what point printed types are truncated.
+// The undocumented defaultMaximumTruncationLength setting determines at what point printed types are truncated in versions less than 5.
 // In kea-typegen output, we NEVER want the types truncated, as that results in a syntax error –
 // "... n more ..." is not valid .d.ts syntax. This is a risk with union types in particular, so we disable truncation.
 // See https://github.com/microsoft/TypeScript/blob/cbb8df/src/compiler/utilities.ts#L563
 // and https://github.com/microsoft/TypeScript/blob/cbb8df/src/compiler/checker.ts#L6331-L6334.
-(ts as any).defaultMaximumTruncationLength = Infinity
+if (!/^5\.\d$/.test(ts.versionMajorMinor)) {
+    (ts as any).defaultMaximumTruncationLength = Infinity
+}
```

Comment:
This makes sense and thank for the fix! Though now I wonder what'll happen when TS 6 comes out. It'll likely work the same way as 5. Thus it might be better here to check that `parseInt(ts.versionMajorMinor.split('.')[0]) < 5` or something like that?

Reply:
That makes sense - fixed it!

Have you had any kind of conversation with the TypeScript team about what they consider to be the proper way to do what this code is trying to do?  You can't be the only person parsing TypeScript ASTs and running into this problem...

Reply:
Thanks! v3.3.3 is out with the fix.

And no, I haven't had any contact with the TS team. Things are mostly stable and working, until some TS upgrade changes things again. 🤷